daemon/logger/jsonfilelog/jsonfilelog.go

```diff
@@ -119,6 +120,7 @@ func ValidateLogOpt(cfg map[string]string) error {
 		switch key {
 		case "max-file":
 		case "max-size":
+		case "compression":
```

Comment:
I would like to not have a flag and just compress... do you see why people would not want this?

I guess there's people violating the rule of everything in `/var/lib/docker` should be private to the engine.

Reply:
After discussing with others, we decided to not compress by default in this driver as it could break people who aren't expecting it.